src/krylov_phiv_adaptive.jl

```diff
@@ -174,6 +179,9 @@ function phiv_timestep!(U::AbstractMatrix{T}, ts::Vector{tType}, A, B::AbstractM
         end
         # Part 2: compute ϕp(tau*A)wp using Krylov, possibly with adaptation
         arnoldi!(Ks, A, @view(W[:, end]); tol = tol, m = m, opnorm = opnorm, iop = iop)
+        if Ks.wasbreakdown
```

Comment:
It seems you're assuming `Ks.wasbreakdown` was false to start `phiv_timestep!` but that's not necessarily true if it's the same cache now reused for a new problem. I think this needs to be set to `false` at the top each time.